PMM-12251 Service Accounts. (#1225)
percona/pmm-doc: docs/api/index.md

> Nodes and Services are external by nature – we do not manage them (create, destroy), but merely maintain a list of them (add to inventory, remove from inventory) in `pmm-managed`. Most Agents are started and stopped by `pmm-agent`. One exception is the External Exporter Type which is started externally.

Reviewer: ```suggestion
Nodes and Services are inherently external; we don't manage their creation or deletion but rather maintain a list of them within pmm-managed by adding them to or removing them from the inventory. The majority of Agents are initiated and halted by pmm-agent, with one exception being the External Exporter Type, which is initiated externally.```

> ## Service accounts authentication

Reviewer: ```suggestion
## Authentication with service accounts
```

!!! caution alert alert-warning "Important"

>     API keys are deprecated now. During update to version PMM v3 all of them you be converted into Service Accounts with Service Token. In such case API key persist same, but it is just covnerted into Service Token.

Reviewer: ```suggestion
Starting with version 3, PMM no longer uses API keys as the primary method for controlling access to the PMM Server components and resources. Instead, PMM is now leveraging Grafana service accounts, which have limited scopes and offer enhanced security compared to API keys.

**Automatic migration of API keys**
When you install PMM 3.x, any existing API keys will be seamlessly converted to service accounts with corresponding service tokens.
	```

> Service accounts are used to control access to the PMM server components and resources. With an API key, you are authenticated to the PMM server, have access to PMM server components and resources, and perform various actions on them. You can use Service accounts as a replacement for basic authentication and API keys.

Reviewer: Service accounts in PMM provide a secure and efficient way to manage access to the PMM Server and its resources. They serve as a replacement for the basic authentication and API keys used in previous versions of PMM (v.2 and earlier).

With service accounts, you can:

- Control access to PMM Server components and resources.
- Define granular permissions for various actions.
- Create and manage multiple access tokens for a single service account.

Creating multiple tokens for the same service account is beneficial in the following scenarios:

- when multiple applications require the same permissions but need to be audited or managed separately. By assigning each application its own token, you can track and control their actions individually.
when a token becomes compromised and needs to be replaced. Instead of revoking the entire service account, you can rotate or replace the affected token without disrupting other applications using the same service account.
- when you want to implement token lifecycle management. You can set expiration dates for individual tokens, ensuring that they are regularly rotated and reducing the risk of unauthorized access.

Reviewer 2: @catalinaadam I think it supposed to be suggestion, right?

Author: Changed.

> Service account contains Service Tokens, which is direct replacement for API key.

Reviewer: ```suggestion
```
